spec.bs

```diff
@@ -478,9 +480,10 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
     [=auction config/per buyer experiment group ids=].
   1. If |perBuyerExperimentGroupIds| is not null and |perBuyerExperimentGroupIds|[|buyer|] exists:
     1. Set |buyerExperimentGroupId| to |perBuyerExperimentGroupIds|[|buyer|].
-  1. TODO: Let |perBuyerSignals| be null/empty
-  1. TODO: If |auctionConfig|'s [=auction config/per buyer signals=][|buyer|] exists:
-    1. TODO: Set |perBuyerSignals| be |auctionConfig|'s [=auction config/per buyer signals=][|buyer|].
+  1. If |auctionConfig|'s [=auction config/per buyer signals=][|buyer|] exists:
+    1. Let |perBuyerSignals| be |auctionConfig|'s [=auction config/per buyer signals=][|buyer|].
+  1. Otherwise:
+    1. Let |perBuyerSignals| be null

```

Comment:
This few steps could be simplified into one if you'd like: `Let |perBuyerSignals| be |auctionConfig|'s [=auction config/per buyer signals=][|buyer|] if that member [=map/exists=], and null otherwise.` That's a fairly common way of doing this sort of thing.